figs/stanplots/analyses_wanang.R

```diff
@@ -34,9 +46,121 @@ l1W_ranef <- brm(log(BIO) ~ TREAT + (1+TREAT|GARDEN),
+                 file="bioAllrenef_lnormal", cores = 4, sample_prior = TRUE)
+
+stancode(l1W_ranef)
+
+#what are the parameters here worth thinking about?
+brms::parnames(l1W_ranef)
+#get the info on divergent iterations
+np <- nuts_params(l1W_ranef)
+# customize appearance of divergences
+color_scheme_set("darkgray")
+div_style <- parcoord_style_np(div_color = "orange", div_size = 1, div_alpha = 1)
+mcmc_parcoord(as.array(l1W_ranef),
+              size = 0.25, alpha = 0.1,
+              regex_pars = "sd_.*",
+              np = np, np_style = div_style)+ coord_flip()
+
+l1W_ranef %>% summary
+
+# recoding factors --------------------------------------------------------
+
+#the model above still has a hard time with divergent iterations. I am curious
+#to try how it works if the variables are coded by hand to develop contrasts
+
+AllTestData_recode <- AllTestData %>% 
+  #remove the weevil stuff just for now
+  filter(!(TREAT %>% str_detect("WEEVIL"))) %>% 
+  # removal treatments as several columns
+  mutate(enemy_removal = if_else(TREAT %in% c("FUNGICIDE", "INSECTICIDE", "PREDATOR"), true = 1, false = 0),
+         is_spray      = if_else(TREAT %in% c("FUNGICIDE", "INSECTICIDE"),             true = 1, false = 0),
+         insecticide   = if_else(TREAT %in% c("INSECTICIDE"),                          true = 1, false = 0))
+
+
+
+bio_lnorm_recode_bf <- bf(BIO ~ 1 + enemy_removal + is_spray + insecticide + (1 + enemy_removal + is_spray + insecticide |GARDEN), family = lognormal())
+brms::get_prior(bio_lnorm_recode_bf, data = AllTestData_recode)
+## Set some priors 
+priors <- c(set_prior("normal(4, 2)", class = "Intercept"),
+            set_prior("normal(0, 3)", class = "b"),
+            set_prior("normal(0, 2)", class = "sd"),
+            set_prior("normal(0, 3)", class = "sigma"),
+            set_prior("lkj(2)", class = "cor"))
+
+## Whole community with random intercept and slope
+bio_recode <- brm(bio_lnorm_recode_bf, 
+                 data=AllTestData_recode,
+                 prior = priors,
+                 control = list(adapt_delta = 0.9),
+                 file="bio_recode", cores = 4, sample_prior = TRUE)
+
+mcmc_parcoord(as.array(bio_recode),
+              size = 0.25, alpha = 0.1,
+              regex_pars = "sd_.*",
+              np = nuts_params(bio_recode), np_style = div_style)+ coord_flip()
+
+# experiment with gamma distribution for biomass instead ------------------
+
+
+#perhaps not necessary, since using lognormal distribution seems to make a big difference.
+
+
+# weevils as a number ------------------------------------------------------
+
+AllTestData_numweevil <- AllTestData %>% 
+  # could use relevel here
+  mutate(removal_trt = if_else(TREAT %in% c("CONTROL", "WEEVIL125", "WEEVIL25"), true = "0_removal", false = as.character(TREAT)),
+         removal_trt = as.factor(removal_trt),
+         weevil = readr::parse_number(TREAT)) %>% 
+  tidyr::replace_na(list(weevil = 0)) %>% 
+  mutate(weevil = weevil / 25)
```

Comment:
Here is the conversion of weevil into numbers! and then scaling it to be a modest number. i don know if you use dplyr, but this is one (clumsy!) example of when it is useful!